config/routes.rb

```diff
@@ -0,0 +1,16 @@
+Rails.application.routes.draw do
+  root to: 'homepages#index'
+
+  resources :works
+  resources :users
+  # get '/users', to: 'users#index', as: 'users'
+
+  get '/login', to: 'users#login_form', as: 'login'
+  post '/login', to: 'users#login'
+  post '/logout', to: 'users#logout', as: 'logout'
+  # get '/users/current', to: 'users#current', as: 'current_user'
+  # get '/users/:id', to: 'users#show', as: 'user'
+
+  post '/upvote/:id', to: 'users#upvote', as: 'upvote'
```

Comment:
I'd suggest making this route `post '/users/:work_id', to: 'users#upvote', as: 'upvote'  So the path isn't so wildly different.